src/python/dxpy/sugar/transfers.py

```diff
@@ -0,0 +1,952 @@
+        compression_level (int): Level of compression between 1 and 9 to compress
+            file to. Specify 1 for gzip --fast and 9 for gzip --best. If not
+            specified, --fast is assumed.
+        return_handler (bool): Whether to return a DXFile handler.
+        project (str): The project ID to upload to, if not the currently selected
+            project.
+        wait_on_close (bool): Whether to block until the file has closed.
+        method (str): Compression method; one of 'gz', 'bz2'.
+        max_part_size (int): Optional, maximum file part size, defaults to project
+            value.
+
+    Returns:
+        DNAnexus link or DXFile pointing to the uploaded file (depending on
+        the value of `return_handler`).
+
+    Raises:
+        ValueError: if compression_level not between 1 and 9
+        CalledProcessError: propogated from run_pipe if called command fails
+    """
+    ext = ".{}".format(method)
+    if remote_filename is None:
+        remote_filename = os.path.basename(filename)
+    if not remote_filename.endswith(ext):
+        remote_filename += ext
+
+    if not remote_folder:
+        remote_folder = "/"
+    elif not remote_folder.startswith("/"):
+        remote_folder = "/{}".format(remote_folder)
+
+    remote_path = "{}/{}".format(remote_folder, remote_filename)
+    if project:
+        remote_path = "{}:{}".format(project, remote_path)
+
+    # check that compression level is valid
+    if compression_level < 1 or compression_level > 9:
+        raise ValueError("Compression level must be between 1 and 9")
+
+    if not max_part_size or max_part_size <= 0:
+        max_part_size = _get_max_part_size(project)
+
+    if filename.endswith(ext):
+        cmd = [
+            "dx", "upload", "--brief", "--buffer-size", str(max_part_size),
+            "--path", remote_filename, filename
+        ]
+        file_id = proc.run_cmd(cmd).out
+    else:
+        exe = "bzip2" if method == "bz2" else "gzip"
+        zip_cmd = [exe, "-{0}".format(compression_level), "-c", filename]
+        upload_cmd = [
+            "dx", "upload", "--brief", "--buffer-size", str(max_part_size),
+            "--path", remote_path
+        ]
+        if wait_on_close:
+            upload_cmd.append("--wait")
+        upload_cmd.append("-")
+        file_id = proc.chain_cmds([zip_cmd, upload_cmd], shell=True).out
+
+    return _wrap_file_id(file_id, return_handler)
+
+
+def archive_and_upload_files(
+    filenames,
+    remote_prefix=None,
+    remote_folder="/",
+    compression_level=1,
+    chdir=None,
+    return_handler=False,
+    project=None,
+    wait_on_close=False,
+    method="gz",
+    max_part_size=None
+):
+    """
+    Archive and upload one or more files.
+
+    Shorthand for running 'tar', 'gzip', and 'dx upload' using subprocess on a
+    list of local files/directories.
+
+    Args:
+        filenames (str or list of str): = Local filenames or directories.
```

Comment:
Change "filenames" to `paths` to indicate the argument passed in can be a file or a directory

Reply:
Done